pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +111,96 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoint reflects the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
+
+	// High level state string: Pending, Standby, Loading, Loaded, FailedToLoad
+	// +kubebuilder:default=Pending
+	ActiveModelState ModelState `json:"activeModelState"`
+	// +kubebuilder:default=""
+	TargetModelState ModelState `json:"targetModelState"`
+
+	// Details of last failure, when load of target model is failed or blocked
+	// +optional
+	LastFailureInfo *FailureInfo `json:"lastFailureInfo,omitempty"`
```

Comment:
Maybe even something could be added to this for when one of the revisions fails?